server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/ICSAttachmentWorkflowTest.java

```diff
@@ -0,0 +1,550 @@
+        }
+    }
+
+    @Ignore("See JIRA issue MAILET-151")
+    @Test
+    public void headersShouldBeFilledOnlyWithOneICalAttachmentWhenMailHasSeveral() throws Exception {
+        MimeMessage message = new MimeMessage(Session
+                .getDefaultInstance(new Properties()));
+        
+        MimeMultipart multiPart = new MimeMultipart();
+        MimeBodyPart part = new MimeBodyPart();
+        part.setText("simple text");
+        multiPart.addBodyPart(part);
+        multiPart.addBodyPart(createAttachmentBodyPart(ICS_1.getBytes(Charsets.UTF_8), "test.txt"));
+        multiPart.addBodyPart(createAttachmentBodyPart(ICS_2.getBytes(Charsets.UTF_8), "test.txt"));
+        multiPart.addBodyPart(createAttachmentBodyPart(ICS_3.getBytes(Charsets.UTF_8), "test.txt"));
+        
+        message.setSubject("test");
+        message.setContent(multiPart);
+        message.saveChanges();
+        
+        Mail mail = FakeMail.builder()
+              .mimeMessage(message)
+              .sender(new MailAddress(FROM))
+              .recipient(new MailAddress(RECIPIENT))
+              .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG);
+                IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(RECIPIENT, PASSWORD));
+
+            String receivedHeaders = imapMessageReader.readFirstMessageHeadersInInbox(RECIPIENT, PASSWORD);
+            
+            //Here only the third ICal attachment is used to fill headers
+            assertThat(receivedHeaders).contains("X-MEETING-UID: " + ICS_UID);
+            assertThat(receivedHeaders).contains("X-MEETING-METHOD: " + ICS_METHOD);
+            assertThat(receivedHeaders).contains("X-MEETING-SEQUENCE: " + ICS_SEQUENCE);
+            assertThat(receivedHeaders).contains("X-MEETING-DTSTAMP: " + ICS_DTSTAMP);
+        }
+    }
+
+
+    private MimeBodyPart createAttachmentBodyPart(byte[] body, String fileName) throws MessagingException, UnsupportedEncodingException {
+        MimeBodyPart part = createBodyPart(body);
+        part.setDisposition("attachment");
+        part.setFileName(fileName);
+        return part;
+    }
+
+    private MimeBodyPart createBodyPart(byte[] body) throws MessagingException, UnsupportedEncodingException {
+        return new MimeBodyPart(new ByteArrayInputStream(
+                Bytes.concat("Content-Transfer-Encoding: 8bit\r\nContent-Type: application/octet-stream; charset=utf-8\r\n\r\n".getBytes(Charsets.UTF_8),
+                        body)));
+    }
+
+    private void waitingForRabbitToBeReady(ConnectionFactory factory) {
```

Comment:
Please introduce a Rabbit MQ Junit rule, and also use it in AmqpForwardAttachmentTest